src/lib/hash/hash.cpp

```diff
@@ -97,6 +101,17 @@ namespace Botan {
 std::unique_ptr<HashFunction> HashFunction::create(const std::string& algo_spec,
                                                    const std::string& provider)
    {
+#if defined(BOTAN_HAS_BEARSSL)
+   if(provider.empty() || provider == "bearssl")
```

Comment:
Agreed.  I put it in front because I felt like sorting it alphabetically, but I will move it after OpenSSL, that's better, yes.